examples/3.routes/8.tolls-and-ferries/interface.js

```diff
@@ -0,0 +1,260 @@
+          `;
+          break;
+        case 'station':
+          instructions = `
+          <div class='timeline-item station'>
+          <div class="icon timeline-icon chargeTurbo station"></div><div class="timeline-content">
+          <div class="station-title-container"><p class='title'>${journeyLegs[i].name}</p></div>
+          <div class='station-info'><p class='body-small metal'>Range at arrival:</p><p>
+          ${journeyLegs[i].rangeEndPercentage}%</p></div>
+          <div class='station-info'><p class='body-small metal'>Available plugs:</p><p>
+          ${journeyLegs[i].plugsAvailable}</p></div>
+          <div class='station-info'><p class='body-small metal'>Charging time:</p><p>
+          ${getDurationString(journeyLegs[i].chargeTime)}</p></div>
+          <div class='station-info'><p class='body-small metal'>Range at departure:</p><p>
+          ${journeyLegs[i + 1].rangeStartPercentage}%</p>
+          </div></div></div>
+          `;
+          break;
+        default:
+          if (journeyLegs[i].tollsOnLeg) {
+            borderCrossingHTML = `<span class='toll'><p> Toll roads</p></span>`;
+          }
+          instructions = `
+          <div class='timeline-item vehicle'>
+          <div class="icon timeline-icon vehicle"></div>
+          <div class='timeline-content'>
+          <div class='drive-title-container'>
+          <p class='title'>Drive ${(journeyLegs[i].distance / 1000).toFixed(1)} km</p>${borderCrossingHTML}</div>
+          <p class='body-small metal'>Approx ${getDurationString(journeyLegs[i].duration)}</p></div></div>
+          `;
+          break;
+      }
+      document.getElementById('journey-timeline').innerHTML += `${instructions}`;
+    }
+    // Show end point
+    document.getElementById('journey-timeline').innerHTML += `<div class="timeline-item timelineDestination end">
+    <div class="icon timeline-icon timelineDestination end"></div>
+    <div class='timeline-content'>
+    <p class='title'>${destination}</p></div></div>`;
+  }
+};
+
+/**
+ * Display ferries and tolls if present
+ *
+ * @return {string} html elements to render ferry/toll information
+ */
+function displayFerriesAndTollsIndicator() {
+  let ferriesAndTollsIconsIndicator = ``;
+  let ferriesAndTollsTextIndicator = ``;
+  if (wholeRouteContainsTolls) {
+    ferriesAndTollsIconsIndicator += `<span class="icon-toll"></span>`;
+    ferriesAndTollsTextIndicator += ` Tolls | `;
+  }
+  if (wholeRouteContainsFerries) {
+    ferriesAndTollsIconsIndicator += `<span class="icon-ferry"></span>`;
+    ferriesAndTollsTextIndicator += `Ferry`;
+  }
+  return `${ferriesAndTollsIconsIndicator} 
+  <h4 id="tolls-indicator"> ${ferriesAndTollsTextIndicator}</h4>`;
+}
+
+/**
+ * Display checks if tolls are present on the leg
+ *
+ * @param leg
+ * @return {boolean}
+ */
+function containsTolls(leg) {
+  for (let i = 0; i < leg.steps.length; i++) {
```

Comment:
You can use the tags property on the route level. 

query getRoute {
  route(id: "63245f4ecb28b0bac7c613a0") {
    route {
      tags
    }
    status
  }
}

Reply:
nice, didn't realise that. this is a better solution for the ```wholeRouteContainsTolls``` variable.
although for splitting the leg up, I think it's still necessary to check on which leg the toll is, as it needs to be displayed on the correct 'drive' part (on the journey overview)

Reply:
the same property exists on a leg level. 
query getRoute {
  route(id: "--FILL-IN-ROUTE-ID--") {
    route {
      tags
      legs{
        tags
      }
    }
    status
  }
}

Reply:
right, but as I understand it correctly, legs coming from the backend are only split up by charging stations.
so a **single** leg could contain:
1. drive to a ferry
2. ferry
3. drive from a ferry to a charging station
4. a charging station.
in this scenario it's necessary to know on which part the toll is on, to display on the journey overview